cpp/oneapi/dal/algo/louvain/vertex_partitioning_types.hpp

```diff
@@ -0,0 +1,138 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+/// @file
+/// Contains the definition of the input and output for the Louvain
+/// algorithm
+
+#pragma once
+
+#include "oneapi/dal/algo/louvain/common.hpp"
+#include "oneapi/dal/algo/louvain/detail/vertex_partitioning_types.hpp"
+
+namespace oneapi::dal::preview::louvain {
+
+/// Class for the description of the input parameters of the Louvain
+/// algorithm
+///
+/// @tparam Graph  Type of the input graph
+template <typename Graph, typename Task = task::by_default>
+class vertex_partitioning_input : public base {
+    static_assert(detail::is_valid_task<Task>);
+
+public:
+    using task_t = Task;
+    // static_assert(detail::is_valid_graph<Graph>,
+    //               "Only undirected_adjacency_vector_graph is supported.");
+    /// Constructs the algorithm input initialized with the graph
+    ///
+    /// @param [in]   g  The input graph
+    vertex_partitioning_input(const Graph &g);
```

Comment:
I thought we have an agreement to have a consistent solution at least now to have one constructor with optional arg via "="